src/codegen/class-definitions.ts

```diff
@@ -44,19 +49,71 @@ export type Field =
        * function: `camelCase(fileName + functionName + "CodeGenerator"`)
        */
       builtin: string;
+      /**
+       * Number of parameters accepted by the function.
+       *
+       * Sets [`function.length`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/length).
+       */
       length?: number;
     };
 
 export class ClassDefinition {
+  /**
+   * Class name.
+   *
+   * Used to find the proper struct and as the `.name` of the JS constructor
+   * function.
+   */
   name: string;
+  /**
+   * Class constructor is newable.
+   */
   construct?: boolean;
+  /**
+   * Class constructor is callable. In JS, ES6 class constructors are not
+   * callable.
+   */
   call?: boolean;
+  /**
+   * ## IMPORTANT
+   * You _must_ free the pointer to your native class!
+   * ```zig
+   * pub const NativeClass = struct {
+   *   pub usingnamespace bun.New(NativeClass);
+   *
+   *   fn constructor(global: *JSC.JSGlobalObject, frame: *JSC.CallFrame) bun.JSError!*SocketAddress {
+   *     // do stuff
+   *     return NativeClass.new(.{
+   *       // ...
+   *     });
+   *   }
+   *
+   *   fn finalize(this: *NativeClass) void {
+   *     // free allocations owned by this class, then free the struct itself.
+   *     this.destroy();
+   *   }
+   * };
+   * ```
+   * @todo remove this and require all classes to implement `finalize`.
+   */
   finalize?: boolean;
   overridesToJS?: boolean;
+  /**
+   * Static properties and methods.
+   */
   klass: Record<string, Field>;
+  /**
+   * properties and methods on the prototype.
+   */
   proto: Record<string, Field>;
+  /**
+   * Properties and methods attached to the instance itself.
+   */
   own: Record<string, string>;
   values?: string[];
+  /**
+   * Set this to `"0b11101110"`.
```

Comment:
Not always, we only want to do that when we will frequently be calling .as() on it